pkg/rbac/sharedData.go

```diff
@@ -32,116 +40,173 @@ type resource struct {
 	kind     string
 }
 
-func (cache *Cache) ClusterScopedResources(ctx context.Context) ([]resource, error) {
-	clusterScoped, err := cache.shared.getClusterScopedResources(cache, ctx)
-	return clusterScoped, err
+var managedClusterResourceGvr = schema.GroupVersionResource{
+	Group:    "cluster.open-cluster-management.io",
+	Version:  "v1",
+	Resource: "managedclusters",
 }
 
-func (shared *SharedData) getClusterScopedResources(cache *Cache, ctx context.Context) ([]resource,
-	error) {
+func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+
+	if sharedCacheValid(&cache.shared) { //if all cache is valid we use cache data
+		klog.V(5).Info("Using shared data from cache.")
+		return nil
+	} else { //get data and cache
+
+		// get all cluster-scoped resources and cache in shared.csResources
+		err := cache.shared.getClusterScopedResources(cache, ctx)
```

Comment:
Use uppercase to match `GetSharedNamespaces` and `GetSharedManagedClusters`
```suggestion
		err := cache.shared.GetClusterScopedResources(cache, ctx)
```